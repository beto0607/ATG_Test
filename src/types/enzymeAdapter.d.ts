// To avoid a typescript error with enzyme-adaoter-react-16
declare module 'enzyme-adapter-react-16';