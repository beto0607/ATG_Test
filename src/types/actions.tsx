import { GameSchedule, GameData } from "../types";

export const GAME_SCHEDULE_LOADED = "GAME_SCHEDULE_LOADED";
export type GAME_SCHEDULE_LOADED = typeof GAME_SCHEDULE_LOADED;

export const GAME_DATA_LOADED = "GAME_DATA_LOADED";
export type GAME_DATA_LOADED = typeof GAME_DATA_LOADED;


export interface GameScheduleLoadedAction {
    type: typeof GAME_SCHEDULE_LOADED;
    text: string;
    gameSchedule?: GameScheduleÂ | null;
}

export interface GameDataLoadedAction {
    type: typeof GAME_DATA_LOADED;
    gameData?: GameData;
}

export type LoadedActions = GameScheduleLoadedAction | GameDataLoadedAction;

export type AppActions = LoadedActions;