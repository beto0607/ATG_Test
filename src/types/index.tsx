type TrackInfo = {
    id: number,
    name: string
};
type FavoriteInfo = {
    raceId: string,
    legNumber: number,
    start: {
        number: number,
        name: string,
        distribution: number
    }
};
type GameInfo = {
    id: string,
    startTime: string, // YYYY-MM-DDThh:mm:ss
    tracks: Array<TrackInfo>,
    favorites: Array<FavoriteInfo>
}
//ENDPOINT RESPONSE
//https://www.atg.se/services/racinginfo/v1/api/products/<gameType>
type GameSchedule = {
    betType: string,
    upcoming: Array<GameInfo>,
    results: Array<GameInfo>
};

type RaceStart = {
    number: number,
    driver: {
        firstName: string,
        lastName: string
    },
    horse: {
        name: string,
        trainer: {
            firstName: string,
            lastName: string
        },
        pedigree: {
            father: {
                name: string
            }
        }
    }
}
type GameRace = {
    date: string,
    name: string,
    scheduledStartTime: string,
    starts: Array<RaceStart>
}
//ENDPOINT RESPONSE
// https://www.atg.se/services/racinginfo/v1/api/games/<gameId>
type GameResponse = {
    id: string,
    status: string,
    races: Array<GameRace>
};

export interface ApplicationState {
    text: string;
    game_schedule: GameSchedule;
    game_data: GameResponse;
}