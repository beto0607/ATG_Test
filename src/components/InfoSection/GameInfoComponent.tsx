import React, { useState } from "react";
import { GameInfo } from '../../types/index';

const GameInfoComponent: React.FC<GameInfo> = ({ id, startTime, tracks, favorites }: GameInfo) => {
    return (
        <li>
            <span>#{id}</span>
        </li>
    );
};

export default GameInfoComponent;
