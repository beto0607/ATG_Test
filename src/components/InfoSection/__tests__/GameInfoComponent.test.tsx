import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import { GameInfo } from '../../../types/index';
import { Collapse, Button } from 'react-bootstrap';
import GameInfoComponent from '../GameInfoComponent';

const mockGameInfo: GameInfo = {
    id: '',
    startTime: "2019-07-24T12:45:47.450Z",
    favorites: [],
    tracks: []
};
describe('getDateFromString ', () => {
    it('should render without crashing', () => {
        const props: GameInfo = mockGameInfo;
        const component = shallow((<GameInfoComponent {...props} />));
        expect(toJson(component)).toMatchSnapshot()
    });
    it('should render collapsed by default', () => {
        const props: GameInfo = mockGameInfo;
        const component = shallow((<GameInfoComponent {...props} />));
        expect(component.find(Collapse).prop('in')).toEqual(false);
    })
    it('shold open when button clicked', () => {
        const props: GameInfo = mockGameInfo;
        const component = shallow((<GameInfoComponent {...props} />));
        component.find(Button).simulate('click');
        expect(component.find(Collapse).prop('in')).toEqual(true);
    })
});
