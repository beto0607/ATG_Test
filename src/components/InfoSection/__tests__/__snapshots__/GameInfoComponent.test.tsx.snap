// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDateFromString  should render without crashing 1`] = `
<Bootstrap(ListGroupItem)
  className="list-group-item"
>
  <Button
    active={false}
    aria-controls="example-collapse-text"
    aria-expanded={false}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Race starts:
     
    <t
      ago={false}
      calendar={false}
      date="2019-07-24T12:45:47.450Z"
      decimal={false}
      element={null}
      filter={[Function]}
      format="HH:mm"
      fromNow={false}
      interval={0}
      local={false}
      onChange={[Function]}
      titleFormat=""
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
      withTitle={false}
    />
  </Button>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div>
      <Connect(GameDataComponentConnected)
        load={false}
        parent_id=""
      />
    </div>
  </Collapse>
</Bootstrap(ListGroupItem)>
`;
